### ISSUE TITLE
Postponed annotation evaluation causes `Annotated` to break

### ISSUE THREAD

**Issue by yrd** [strawberry-graphql/strawberry#1819]
When using postponed annotation evaluation, annotating resolver arguments no longer works:

```python
from __future__ import annotations

import random
from typing import Annotated

import strawberry


@strawberry.type
class Query:
    @strawberry.field
    def dice_roll(
        self,
        sides: Annotated[
            int,
            strawberry.argument(description="Number of sides the die should have."),
        ] = 6,
    ) -> int:
        return random.randint(1, sides)


strawberry.Schema(query=Query)
```

The example above raises this TypeError:

```
TypeError: Query fields cannot be resolved. Unexpected type 'typing.Annotated[int, <strawberry.arguments.StrawberryArgumentAnnotation object at 0x7fd12e130d00>]'
```

When the first line (`from __future__ import annotations`) is left out, everything works as intended. This will probably also break once Python 3.11 lands, since the behavior will become mandatory then. #1586 refers to a somewhat related issue.

**Comment by patrick91** [comment]
@yrd I think this has been fixed, I'll close for now, but let me know if is not fixed. 

I tested it here: https://play.strawberry.rocks/?gist=1f311d6ce6b63ef50bdb55115551fbb5
